### PR TITLE
Replace miniconda by miniforge to avoid licensing issues

### PR DESCRIPTION
**Pull request by xavierabellan**
Using miniconda from Anaconda implies accepting its license. This may not be obvious to the user using this tool since it is done automatically, and they may be indadvertedly in breach of such license. 

Replacing miniconda by miniforge is a safer alternative, addressing this licensing concern while virtually keeping all the functionality intact.

### CODE REVIEW

**Comment by Nortamo** [comment]
Overall looks good, I'll have to think if it's better to nuke miniconda all together or then default to miniforge and have an option for using miniconda while removing the `-b` as to show the license to the user

File: templates/conda.sh [hunk 1]
```diff
@@ -32,7 +33,6 @@ print_info "Creating env, full log in $CW_BUILD_TMPDIR/build.log" 1
 
 if [[ ${CW_MAMBA} == "yes" ]] ;then
     print_info "Using mamba to install packages" 1 
-    conda install -y mamba -n base -c conda-forge
```
[the remark this answers is not in the document]
Comment (xavierabellan) [reply templates/conda.sh:35]: miniforge comes with both conda and mamba pre-installed.

**Comment by Nortamo** [comment]
Looks good, I'll leave it to later to possibly re-implement the support for using miniconda. 
Before merging: 

Whats the specific reason for the empty condarc, as that now breaks the functionality that we disable `$HOME/.conda/pkgs`
(if the file is empty but exists the user pkgs setting will be inserted, which is not what we want)

**Comment by xavierabellan** [comment]
Thanks, that's a very good idea. I have committed and pushed the proposed change.

**Comment by xavierabellan** [comment]
I also realised that the environments created within the container are leaked into the user's ~/.conda/environments.txt outside the container. If conda is used outside, all those environments are listed but are, of course, not there.

I would propose to add the following dummy overlay to address that problem using SINGULARITY_BIND:
```
$ git diff
diff --git a/create_inst.sh b/create_inst.sh
index 47a5ab2..a5ccc83 100755
--- a/create_inst.sh
+++ b/create_inst.sh
@@ -42,6 +42,8 @@ SINGULARITY_BIND="$SINGULARITY_BIND,/tmp"
 if [[ ! ${CW_ENABLE_CONDARC+defined} ]]; then
     export CONDA_PKGS_DIRS=$CW_INSTALLATION_PATH/miniforge/pkgs
 fi
+mkdir -p $CW_BUILD_TMPDIR/.conda_override
+SINGULARITY_BIND="$SINGULARITY_BIND,$CW_BUILD_TMPDIR/.conda_override:$(readlink -f $HOME/.conda)"
 export SINGULARITY_BIND
 echo "export install_root=$CW_INSTALLATION_PATH" >> _extra_envs.sh
 echo "export install_root=$CW_INSTALLATION_PATH" >> _vars.sh
```
If you agree, I can also push this change (or open a separate pull request for it)

**Comment by Nortamo** [comment]
> I also realised that the environments created within the container are leaked into the user's ~/.conda/environments.txt outside the container. If conda is used outside, all those environments are listed but are, of course, not there.
> 
> I would propose to add the following dummy overlay to address that problem using SINGULARITY_BIND:
> 
> ```
> $ git diff
> diff --git a/create_inst.sh b/create_inst.sh
> index 47a5ab2..a5ccc83 100755
> --- a/create_inst.sh
> +++ b/create_inst.sh
> @@ -42,6 +42,8 @@ SINGULARITY_BIND="$SINGULARITY_BIND,/tmp"
>  if [[ ! ${CW_ENABLE_CONDARC+defined} ]]; then
>      export CONDA_PKGS_DIRS=$CW_INSTALLATION_PATH/miniforge/pkgs
>  fi
> +mkdir -p $CW_BUILD_TMPDIR/.conda_override
> +SINGULARITY_BIND="$SINGULARITY_BIND,$CW_BUILD_TMPDIR/.conda_override:$(readlink -f $HOME/.conda)"
>  export SINGULARITY_BIND
>  echo "export install_root=$CW_INSTALLATION_PATH" >> _extra_envs.sh
>  echo "export install_root=$CW_INSTALLATION_PATH" >> _vars.sh
> ```
> 
> If you agree, I can also push this change (or open a separate pull request for it)

It can be part of this PR, so you can push the changes here.

**Comment by xavierabellan** [comment]
Done thanks!

**Pull request merged**